src/guiFormSpecMenu.cpp

```diff
@@ -1114,29 +1115,31 @@ void GUIFormSpecMenu::parseTextArea(parserData* data, std::vector<std::string>&
 		258+m_fields.size()
 	);
 
-	if (name.empty()) {
-		// spec field id to 0, this stops submit searching for a value that isn't there
-		addStaticText(Environment, spec.flabel.c_str(), rect, false, true, this, spec.fid);
-	} else {
-		spec.send = true;
+	bool is_editable = !name.empty();
 
+	if (is_editable) {
+		spec.send = true;
+	}
+	
 		gui::IGUIEditBox *e;
 #if USE_FREETYPE && IRRLICHT_VERSION_MAJOR == 1 && IRRLICHT_VERSION_MINOR < 9
-		if (g_settings->getBool("freetype")) {
-			e = (gui::IGUIEditBox *) new gui::intlGUIEditBox(spec.fdefault.c_str(),
-				true, Environment, this, spec.fid, rect);
-			e->drop();
-		} else {
+	if (g_settings->getBool("freetype")) {
+		e = (gui::IGUIEditBox *) new gui::intlGUIEditBox(spec.flabel.c_str(),
+			true, Environment, this, spec.fid, rect, is_editable, true);
+		e->drop();
+	} else {
 #else
-		{
+	{
 #endif
-			e = Environment->addEditBox(spec.fdefault.c_str(), rect, true, this, spec.fid);
-		}
+		e = new GUIEditBoxWithScrollBar(spec.flabel.c_str(), true,
```

Comment:
`spec.fdefault.c_str()` here too. The label of the textarea is not the default content. This mistake here makes editing written signs and books impossible.

Reply:
Well If I change them, there is no text displayed anymore

Reply:
Works for me. 
![grafik](https://user-images.githubusercontent.com/1497498/31318457-65a038ce-ac53-11e7-9896-fafd188d6a40.png)

`spec.flabel.c_src()` will only show the label of the textbox as content instead of the saved sign text.

Reply:
Odd. But ur right, there is still something wrong.
Before: 
![grafik](https://user-images.githubusercontent.com/3142688/31319287-cc672cea-ac60-11e7-987b-f3b61a6e1b36.png)
Url+port not displayed, description displayed.

After:
![grafik](https://user-images.githubusercontent.com/3142688/31319264-44683064-ac60-11e7-8c44-cd25952c47cf.png)
Url+port displayed, description not.

Reply:
Because the server description abuses the textarea and actually uses the label to keep the text within a specified area.
With this PR, the label is only shown for editable textareas (see code lines below) - which is the reason why it no longer shows. Removing the check makes it appear again, but with a nasty offset.

Reply:
Yes. I have not found a place in documentation, that says that such abusement is ok.
If you say that abusement is not right, ill fix the mainmenu and pray that not any mod is doing the same.

Reply:
I'm afraid there's no other size limited read-only formspec element, so it would be best to extend the docs to document this possibility too.

Reply:
I fixed the mainmenu and updated the lua api.